lib/resolve.js

```diff
@@ -42,6 +42,35 @@ const memoizePromise = function (func, resolver) {
 
 exports.addon = memoizePromise(addonResolver, (_, app, id) => `${app}|${id}`)
 
+function NotFound () {
+  Error.call(this)
+  Error.captureStackTrace(this, this.constructor)
+  this.name = this.constructor.name
+
+  this.statusCode = 404
+  this.message = 'Couldn\'t find that addon.'
+}
+
+function AmbiguousError (objects) {
+  Error.call(this)
+  Error.captureStackTrace(this, this.constructor)
+  this.name = this.constructor.name
+
+  this.statusCode = 422
+  this.message = `Ambiguous identifier; multiple matching add-ons found: ${objects.map((object) => object.name).join(', ')}.`
+  this.body = {'id': 'multiple_matches', 'message': this.message}
+}
+
+const singularize = function (matches) {
+  if (matches.length === 0) {
```

Comment:
@dickeyxxx updated in https://github.com/heroku/heroku-cli-addons/pull/55/commits/60ff8b523a2ed188c6d5709e7a129fe48b12c254